### PR TITLE
WIP - consolidate env handling into pkg/env

### PR DESCRIPTION
**Pull request by vrothberg**
Env-variable related code is scattered across several packages making it
hard to maintain and extend.  Consolidate the code into a new pkg/env
package.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
Still want to move the env-parsing code into pkg/env but for now I want to throw CI at it.

**Comment by vrothberg** [comment]
Fixes: #5255

File: libpod/oci_conmon_linux.go [hunk 1]
```diff
@@ -1274,15 +1280,26 @@ func prepareProcessExec(c *Container, cmd, env []string, tty bool, cwd, user, se
...
 
-	hasHomeSet := false
-	for _, s := range pspec.Env {
-		if strings.HasPrefix(s, "HOME=") {
-			hasHomeSet = true
-			break
-		}
+	// We need to start with a fresh environment and should not inherit from the
+	// container's config (https://github.com/containers/libpod/issues/5255).
+	imageData, err := getImageDataFromContainer(c)
```
Comment (mheon) [inline libpod/oci_conmon_linux.go]: Do we need this? We have the full list of environment variables the container was created from in the OCI spec.

Comment (vrothberg) [reply libpod/oci_conmon_linux.go]: According to #5255 we need that to match Docker which always starts with the default + from the image.

Comment (mheon) [reply libpod/oci_conmon_linux.go]: We don't want to use the extra environment variables set by the container process, but we do want those set at container creation time.

Tested with:
Terminal 1 - `sudo docker run -t -i --rm --name test --env a=b fedora bash`
Terminal 2 - `sudo docker exec -t -i test env`

I see `a=b` in the output of exec - so we should be including everything from container create time, but nothing that was added after the container was run.

**Comment by mheon** [comment]
Note to self: since Valentin is off on vacation after today, I'll take this one and finish it up.

**Comment by mheon** [comment]
Closing in favor of #5270

**Pull request closed without merging**